ejb3/src/main/java/org/jboss/as/ejb3/timerservice/TimerServiceImpl.java

```diff
@@ -408,20 +408,18 @@ public TimerImpl loadAutoTimer(ScheduleExpression schedule,
         synchronized (this.timers) {
             for (final TimerImpl timer : this.timers.values()) {
                 // Less disruptive way to get WFLY-8457 fixed.
-                if (timer.isActive() || (!timer.isActive() && timer.getState() == TimerState.ACTIVE)) {
-                    if (timer.getPrimaryKey() == null || timer.getPrimaryKey().equals(pk)) {
-                        activeTimers.add(timer);
-                    }
+                if ((timer.isActive() || !timer.isActive() && timer.getState() == TimerState.ACTIVE)
+                        && (timer.getPrimaryKey() == null || timer.getPrimaryKey().equals(pk))) {
+                    activeTimers.add(timer);
                 }
```

Comment:
The first segment of the above if condition:
`(timer.isActive() || !timer.isActive() && timer.getState() == TimerState.ACTIVE)` can be simplified to:
`(timer.isActive() || timer.getState() == TimerState.ACTIVE)`

Reply:
@chengfang Hello Cheng, I have fixed it. May you review and approve, please?